RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,11 @@
+# Release procedure
+
+- Update [version_history.md](docs/root/version_history.md). 
+  - Make sure breaking changes are explicitly called out.
+  - Ensure the release notes are complete.
+- Tag the release in git using the current version number:
+```bash
+git tag -a "v0.3" -m "Release v0.3"
+git push origin v0.3
```

Comment:
Do we need to do the tagging? If we do a GitHub tagged release, does it generate these for us?

Reply:
Those currently exist, so if that is what you did earlier, then yes :-) I'll amend